Ylaki Sprint 4

ylaki opened the pull request.

ylaki committed:
New commit

pavel-ulan commented on pom.xml:
```diff
@@ -0,0 +1,47 @@
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <version>5.8.2</version>
```
**можно лучше**
вообще вроде застейтили, что используем 4-ю версию (сейчас - ок, но лучше придерживайся правил)

ylaki replied:
JUnit 4 версия стоит

pavel-ulan commented on src/test/java/ImportantQuestionsList.java:
```diff
@@ -0,0 +1,56 @@
+                    "Да, пока самокат не привезли. Штрафа не будет, объяснительной записки тоже не попросим. Все же свои.",
+                    "Да, обязательно. Всем самокатов! И Москве, и Московской области."};
+            assertEquals(expectedQuestion[i], mainPage.openQuestion(i));
+            assertEquals(expectedAnswer[i], mainPage.openAnswer(i));
```
**надо исправить**
все очень даже неплохо начиналось, но тут - нет
вспомни, есть такой подход как параметризованное тестирование (и несколько реализаций). Так вот данный тест, это идеальный кандидат для этого.
Просто для инфы: если тебе надо проверить какой-то типовой пул данных, то как правило выбирается параметризованное тестирование. Циклы для этого никогда не используются. Во первых, т.к. это не дает в отчете правильного результата, во вторых, т.к. любой фейл останавливает выполнение теста (и цикла соответственно) и тебе не будут известны результаты оставшихся сравнений

ylaki replied:
Параметризация сделана.

pavel-ulan commented on src/test/java/PositiveOrderParam.java:
```diff
@@ -0,0 +1,101 @@
+            webDriver.manage().timeouts().implicitlyWait(Duration.of(3, SECONDS));
+
+            //Maximize current window
+            webDriver.manage().window().maximize();
```
**инфо**
надеюсь, обратила внимание, что это деприкейтнутый метод
Это не значит, что его нельзя использовать, но все же желательно избегать

ylaki replied:
IDEA не жалуется на deprecated метод. Уточните, пожалуйста. Наставник тоже не в курсе.

pavel-ulan replied:
Да, мой косяк, сорри. Вероятно с чем-то спутал :)

pavel-ulan reviewed (changes requested):
Неплохо, но тест надо изменить на параметризованный

ylaki committed:
update pom.xml
Parametrized questions
Not clear about deprecated method. Vovk Alexey doesn't know.

pavel-ulan reviewed (approved):
Да, валидно. Молодцом
